src/service/extensions-impl.js

```diff
@@ -452,27 +454,29 @@ export class Extensions {
 /**
  * Calculate script url for amp-ad.
  * @visibleForTesting
+ * @param {string} protocol Serving protocol
  * @param {string} path Location path of the window
+ * @param {string} port Serving port
  * @param {string} extensionId
  * @param {boolean=} isTest
  * @param {boolean=} isUsingCompiledJs
  * @return {string}
  * @visibleForTesting
  */
-export function calculateExtensionScriptUrl(path, extensionId, isTest,
-    isUsingCompiledJs) {
+export function calculateExtensionScriptUrl(protocol, path, port, extensionId,
+    isTest, isUsingCompiledJs) {
   if (getMode().localDev) {
     if (isTest) {
       if (isUsingCompiledJs) {
         return `/base/dist/v0/${extensionId}-0.1.js`;
       }
       return `/base/dist/v0/${extensionId}-0.1.max.js`;
     }
-    if (path.indexOf('.max') >= 0) {
-      return `http://localhost:8000/dist/v0/${extensionId}-0.1.max.js`;
+    if (path.indexOf('.max') >= 0 || path.substr(0, 5) == '/max/') {
+      return `${protocol}//localhost:${port}/dist/v0/${extensionId}-0.1.max.js`;
```

Comment:
Ok, changed it.  Passes tests (on my machine) and seems to work in local testing...  At least for .max paths.  .min paths, for some reason, are still loading the extension from the CDN, and it looks like they're skipping this code path altogether.  I _think_ I did `gulp dist` first, so I'm not sure where they're getting that logic.

Any ideas?

Other than that (which might predate this change -- I haven't verified), I think this is ready to go.